Assets/Scripts/NPCs/StateMachine/States/Chase.cs

```diff
@@ -1,103 +1,123 @@
+        {
+            Vector3 velocity = ComputeChaseVelocity();
 
-        Vector3 currentVelocity = _npcManager.Rigidbody.velocity;
+            //The NPC will always looking where it needs to go.
+            float anglePlayerNpc = NpcManager.LookWhereYouAreGoing(velocity);
 
-        Vector3 steering = desiredVelocity - currentVelocity;
+            //If the player is out FOV, NPC stop moving and rotate until the player is back in FOV
+            if (anglePlayerNpc > NpcGlobalVariables.FieldOfView &&
+                NpcManager.GetDistanceWithPlayer() > NpcGlobalVariables.ChasePlayerRange)
+            {
+                velocity = Vector3.zero;
+            }
 
-        Vector3 velocity = currentVelocity + steering;
+            //The NPC always walk forward.
+            NpcManager.SetAnimatorVelocity(Vector3.forward * velocity.magnitude);
+        }
 
-        if (Vector3.Magnitude(velocity) > _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseMaxVelocity))
+        private Vector3 ComputeChaseVelocity()
         {
-            velocity = (velocity / Vector3.Magnitude(velocity)) *
-                       _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseMaxVelocity);
-        }
+            
+            ComputeCurrentTargetLocation();
 
-        velocity.y = 0;
+            Vector3 desiredVelocity = Vector3.Normalize(_currentTargetLocation - NpcManager.transform.position) *
+                                      NpcManager.GetModelSpeed(NpcGlobalVariables.ChaseAcceleration);
 
-        return velocity;
-    }
+            Vector3 currentVelocity = NpcManager.Rigidbody.velocity;
 
-    private void ComputeAStar()
-    {
-        Vector3 nodeWithoutY = Vector3.zero;
-        if (_currentTargetLocation != NPCsGlobalVariables.DefaultInitialVector)
-        {
-            nodeWithoutY = new Vector3(_currentTargetLocation.x, 0, _currentTargetLocation.z);
+            Vector3 steering = desiredVelocity - currentVelocity;
+
+            var velocity = ComputeVelocity(currentVelocity, steering);
+
+            return velocity;
         }
 
-        if (_currentTargetLocation == NPCsGlobalVariables.DefaultInitialVector ||
-            Vector3.Distance(_npcManager.transform.position, nodeWithoutY) <
-            NPCsGlobalVariables.WithinTargetNodeRange) // TODO Add check if we can see target node
+        private void ComputeAStarForSteering()
         {
-            Node npcNode = NavigationGraph.GetClosestNode(_npcManager.transform.position);
+            Node npcNode = NavigationGraph.GetClosestNode(NpcManager.transform.position);
             Node playerNode = NavigationGraph.GetClosestNode(_playerData.LastSeenPosition);
 
             List<Node> path = Pathfinding.Astar(npcNode, playerNode);
 
-            if (path.Count < 2)
+            if (path == null)
+            {
+                Debug.Log(npcNode);
+            }
+
+            _currentTargetLocation = path[NpcGlobalVariables.NextElementInPath].position;
```

Comment:
The first node in the `path` returned by the Astar is the closest node to the NPC, and that make sense to not go to this node since sometimes it would involve going backwards to reach the node (and then eventually getting stuck there)

But sometimes the 1st node in the path might be usefull to reach first : 
Imagine the following situation, the player is on a side of a house, the NPC is on the other side. The first node in the path is at the corner of the house, and the next one is behind the corner. In this scenario, the NPC would try to run trhough the wall trying to reach `path[NpcGlobalVariables.NextElementInPath]` since it's behind the house's corner

Instead, check if the next node can be seen. If not then go to the first node since it's garenteed from there the NPC will be able to see the next node.

Suggestion : 
```c#
if (Node.CanSeeNode(path[NpcGlobalVariables.NextElementInPath]) {
            _currentTargetLocation = path[NpcGlobalVariables.NextElementInPath].position;
} else {
            _currentTargetLocation = path[0].position;
}
```

This way, we are still targeting the 2nd node in the path most of the time, but when we can't reach it directly, we follow the path from the begining

Reply:
I think that this will not fix the issue, because if the next node is behind a barrel, for instance, the NPC will be stuck on their current node forever. Same can be said about any next node that is not seen by the NPC. However, we have collision avoidance to prevent this issue from happening